lib/gssapi/ntlm/accept_sec_context.c

```diff
@@ -54,10 +54,13 @@ _gss_ntlm_allocate_ctx(OM_uint32 *minor_status, ntlm_ctx *ctx)
     (*ctx)->server = ns_interface;
 
     maj_stat = (*(*ctx)->server->nsi_init)(minor_status, &(*ctx)->ictx);
-    if (maj_stat != GSS_S_COMPLETE)
-	return maj_stat;
+    if (maj_stat == GSS_S_COMPLETE)
+    	return GSS_S_COMPLETE;
 
-    return GSS_S_COMPLETE;
+    if (*ctx) 
+	free(*ctx);
```

Comment:
Set *ctx = NULL after it has been freed.

Reply:
or GSS_C_NO_CONTEXT